indexing-service/src/main/java/org/apache/druid/indexing/common/task/IndexTask.java

```diff
@@ -1196,16 +1190,9 @@ public InputFormat getInputFormat()
       return inputFormat;
     }
 
-    public InputSource getNonNullInputSource(@Nullable InputRowParser inputRowParser)
+    public InputSource getNonNullInputSource()
```

Comment:
this too needs a fix similar to the one you did in SeekableStreamSampler. We need to keep the adapters around.

Reply:
This might not be needed since, from what I observed either we can mention `ioConfig#firehoseFactory + dataSchema#parser` or `ioConfig#inputSource + ioConfig#inputFormat` combination in the ingestion spec, now that FiniteFirehoseFactory is removed user has to shift to using `ioConfig#inputSource + ioConfig#inputFormat`. In the case of `SeekableStreamSamplerSpec`, adapters have to be used since users can submit either `ioConfig#inputFormat` or `dataSchema#parser`, and `FirehoseFactory` or `InputSource` isn't submitted in `SeekableStreamIndexTaskIOConfig`.